Add note on how cargo build-bpf works (#30938)
solana-labs/solana: docs/src/getstarted/rust.md

> **NOTE:**
> After each time you build your Solana program, the above command will output the build path of your compiled program's `.so` file and the default keyfile that will be used for the program's address.

> > cargo build-bpf will install the toolchain from the installed solana CLI tools. One may need to upgrade that if any version incompatibilities are encountered.

Reviewer: ```suggestion
> `cargo build-bpf` installs the toolchain from the currently installed solana CLI tools. You may need to upgrade those tools if you encounter any version incompatibilities.
```

Author: cool. I updated it to that. Thanks
